### ISSUE TITLE
Anomaly "Evar ?X62 was not declared" on `case H: eqt` with `eqt: X -> X -> bool` being a function

### ISSUE THREAD

**Issue by ComFreek** [coq/coq#10179]
#### Description of the problem

Might be related to #10175: `case H: eqt` throws an anomaly where `eqt: X -> X -> bool` is a function.

```coq
From Coq Require Import ssreflect ssrbool.
From Coq Require Import Lists.List.
Import ListNotations.

Fixpoint contains {X: Type} (eqt: X -> X -> bool) (l: list X) (x: X): bool := true.

Fixpoint deduplicate {X: Type} (eqt: X -> X -> bool) (l: list X): list X := match l with
| nil => nil
| x :: xs => let deduplicatedRest := deduplicate eqt xs in
  if (contains eqt deduplicatedRest x) then deduplicatedRest
  else x :: deduplicatedRest
end.

Theorem deduplicate_idempotent: forall {X: Type} (eqt: X -> X -> bool) (l: list X), deduplicate eqt (deduplicate eqt l) = deduplicate eqt l.
Proof.
move => ? eqt.
elim => [|x xs IHxs] => //.
simpl.

case H: eqt.
```

Error message:

> Anomaly "Evar ?X62 was not declared."
> Please report at http://coq.inria.fr/bugs/.

Note that

```coq
(* Code same as above including the simpl line *)
simpl.

(* clear *)
case H: contains eqt (deduplicate eqt xs) x.
```
throws no anomaly errors (but the tactic fails anyway as expected), however, with `clear` commented back in again in throws an anomaly "Evar ?X153 was not declared".

#### Coq Version

Coq 8.9.0 on Windows 10 Pro 64-bit.

Details:

> The Coq Proof Assistant, version 8.9.0 (January 2019)
compiled on Jan 17 2019 16:07:40 with OCaml 4.07.0

**Comment by LasseBlaauwbroek** [comment]
This appears to have been fixed in v8.11. Recommend closing. CC @gares as someone who could confirm.

**Comment by SkySkimmer** [comment]
@LasseBlaauwbroek I can reproduce on 8.16 but not master, did you mean it seems to have been fixed in 8.17?
